lib-multisrc/mmlook/src/eu/kanade/tachiyomi/multisrc/mmlook/MMLook.kt

```diff
@@ -0,0 +1,200 @@
+        CategoryFilter(),
+    )
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotBlank()) {
+            return POST(
+                "$desktopUrl/s",
+                headers,
+                FormBody.Builder().add("k", query.take(12)).build(),
+            )
+        }
+        for (filter in filters) {
+            when (filter) {
+                is RankingFilter -> if (filter.state > 0) {
+                    return rankingRequest(filter.options[filter.state].value)
+                }
+
+                is CategoryFilter -> if (filter.state > 0) {
+                    val id = filter.options[filter.state].value
+                    return GET("$desktopUrl/sort/$id", headers)
+                }
+
+                else -> {}
+            }
+        }
+        return popularMangaRequest(page)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        if (response.request.method == "GET") return popularMangaParse(response)
+
+        val entries = response.asJsoup().select(".col-auto").map { element ->
+            SManga.create().apply {
+                url = element.selectFirst("a")!!.attr("href").mustRemoveSurrounding("/", "/")
+                title = element.selectFirst(".e-title")!!.text()
+                author = element.selectFirst(".tip")!!.text()
+                thumbnail_url = element.selectFirst("img")!!.attr("data-src")
+            }.formatUrl()
+        }
+        return MangasPage(entries, false)
+    }
+
+    override fun getMangaUrl(manga: SManga): String {
+        val id = manga.url.removeSurrounding("/")
+        return "$baseUrl/$id/".certificateWorkaround()
+    }
+
+    // Desktop page has consistent template and more initial chapters
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val id = manga.url.removeSurrounding("/")
+        return GET("$desktopUrl/$id/", headers)
+    }
+
+    override fun mangaDetailsParse(response: Response) = SManga.create().apply {
+        val comicInfo = response.asJsoup().selectFirst(".comicInfo")!!
+        thumbnail_url = comicInfo.selectFirst("img")!!.attr("data-src")
+
+        val container = comicInfo.selectFirst(".detinfo")!!
+        title = container.selectFirst("h1")!!.text()
+
+        var updated = ""
+        for (span in container.select("span")) {
+            val text = span.ownText()
+            val value = text.substring(4).trimStart()
+            when (val key = text.substring(0, 4)) {
+                "作 者：" -> author = value
+                "更新时间" -> updated = "$text\n\n"
+                "标 签：" -> genre = value.replace(" ", ", ")
+                "状 态：" -> status = when (value) {
+                    "连载中" -> SManga.ONGOING
+                    "已完结" -> SManga.COMPLETED
+                    else -> SManga.UNKNOWN
+                }
+
+                else -> throw Exception("Unknown field: $key")
```

Comment:
don't throw?

Reply:
Same above. This makes website changes easier to notice. Being stricter is not a bad thing here.